cirq/ops/common_channels.py

```diff
@@ -446,6 +446,67 @@ def amplitude_damp(gamma: float) -> AmplitudeDampingChannel:
     return AmplitudeDampingChannel(gamma)
 
 
+@value.value_equality
+class ResetChannel(gate_features.SingleQubitGate):
```

Comment:
It's not clear that there isn't a need for a more general reset channel that resets to states different that |0>.  This is often used in QEC.

Reply:
I can implement this in a following PR. Would resetting to 0, 1, +, -, etc be enough or are you envisioning reseting to arbitrary single qubit states? For full control I guess one would like to be able to specify that a measured 0 resets to X and a measured 1 goes to Y, where X and Y are some arbitrary states.